src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetBuildInvocation.cs

```diff
@@ -84,7 +88,46 @@ public void MsbuildInvocationIsCorrect(string[] args, string expectedAdditionalA
                     .Should()
                     .Be($"{ExpectedPrefix} -nologo -consoleloggerparameters:Summary{expectedAdditionalArgs}");
             });
+        }
+
+        [Fact]
+        public void OsOptionIsCorrectlyResolved()
+        {
+            CommandDirectoryContext.PerformActionWithBasePath(WorkingDirectory, () =>
+            {
+                var msbuildPath = "<msbuildpath>";
+                var command = BuildCommand.FromArgs(new string[] { "--os", "os" }, msbuildPath);
+                var expectedArch = Environment.Is64BitOperatingSystem ? "x64" : "x86";
+                command.GetArgumentsToMSBuild()
+                    .Should()
+                    .Be($"{ExpectedPrefix} -restore -consoleloggerparameters:Summary -property:RuntimeIdentifier=os-{expectedArch}");
+            });
+        }
+
+        [WindowsOnlyFact]
```

Comment:
Can we support the test on all platforms? Specifically macOS as that is currently the most important platform where this has to work.